sklearn/model_selection/_plot.py

```diff
@@ -451,3 +459,452 @@ def from_estimator(
+            `None`, the generic name "Score" will be used.
+
+        score_type : {"test", "train", "both"}, default="test"
+            The type of score to plot. Can be one of `"test"`, `"train"`, or
+            `"both"`.
+
+        log_scale : bool, default=False
+            Whether or not to use a logarithmic scale for the x-axis.
+
+        std_display_style : {"errorbar", "fill_between"} or None, default="fill_between"
+            The style used to display the score standard deviation around the
+            mean score. If None, no standard deviation representation is
+            displayed.
+
+        line_kw : dict, default=None
+            Additional keyword arguments passed to the `plt.plot` used to draw
+            the mean score.
+
+        fill_between_kw : dict, default=None
+            Additional keyword arguments passed to the `plt.fill_between` used
+            to draw the score standard deviation.
+
+        errorbar_kw : dict, default=None
+            Additional keyword arguments passed to the `plt.errorbar` used to
+            draw mean score and standard deviation score.
+
+        Returns
+        -------
+        display : :class:`~sklearn.model_selection.ValidationCurveDisplay`
+            Object that stores computed values.
+        """
+        check_matplotlib_support(f"{self.__class__.__name__}.plot")
+
+        import matplotlib.pyplot as plt
+
+        if ax is None:
+            _, ax = plt.subplots()
+
+        if negate_score:
+            train_scores, test_scores = -self.train_scores, -self.test_scores
+        else:
+            train_scores, test_scores = self.train_scores, self.test_scores
+
+        if std_display_style not in ("errorbar", "fill_between", None):
+            raise ValueError(
+                f"Unknown std_display_style: {std_display_style}. Should be one of"
+                " 'errorbar', 'fill_between', or None."
+            )
+
+        if score_type not in ("test", "train", "both"):
+            raise ValueError(
+                f"Unknown score_type: {score_type}. Should be one of 'test', "
+                "'train', or 'both'."
+            )
+
+        if score_type == "train":
+            scores = {"Training metric": train_scores}
+        elif score_type == "test":
+            scores = {"Testing metric": test_scores}
+        else:  # score_type == "both"
+            scores = {"Training metric": train_scores, "Testing metric": test_scores}
+
+        if std_display_style in ("fill_between", None):
+            # plot the mean score
+            if line_kw is None:
+                line_kw = {}
+
+            self.lines_ = []
+            for line_label, score in scores.items():
+                self.lines_.append(
+                    *ax.plot(
+                        self.param_range,
+                        score.mean(axis=1),
+                        label=line_label,
+                        **line_kw,
+                    )
+                )
```

Comment:
The label here will be ``Training metric`` or ``testing metric``. I think it would be more user friendly to give the actual name of the metric

Reply:
I recall that we had a similar discussion on the `LearningCurveDisplay`. From what I recall it is not necessarily easy to extract the score name: the fact that we have the `negate_score` with some scores being errors and not all names being consistent makes it difficult to infer the score name.

Reply:
Since we pass the `score_name`, we could show that, with a `neg` prefix if `negate_score` ?
The y axis label shows the score_name btw even if `negate_score`. Should we prefix it by `neg` ?

Reply:
OK I will try to provide something sensible.

Reply:
Actually, I am wondering if `train` and `test` are sufficient labels.
The information regarding the score should appear on the y-axis label.

Reply:
I'm okay with that. I still think that the `neg` prefix should be added when `negate_score` is True.